EXPLAINER.md

```diff
@@ -262,154 +298,225 @@ choosing appropriate window placements. The proposed shape of this query is a
-                         // Useful for placing control panels on touch-screens.
+  // NEW: Requests permission-gated access to additional screen information.
+  [SecureContext] Promise<Screens> getScreens();
 };
 ```
 
 This method gives the web platform a surface to optionally expose an appropriate
 amount of multi-screen information to web applications. By returning a promise,
 user agents can asynchronously determine what amount of information to expose,
-prompt users to decide, calculate the resulting values lazily, and reject or
+prompt users to decide, obtain underlying information lazily, and reject or
 resolve accordingly.
 
-`ScreenInfo` objects retrieved from `getScreens()` are integral for multi-screen
-window placement use cases. The relative bounds establish a coordinate system
-for cross-screen window placement, while the newly exposed properties of the
-display devices allow applications to restore or choose window placements.
+The slideshow example can now request access to multi-screen information:
+
+```js
+// Request information for multi-screen slideshow functionality, if available.
+async function startSlideshow() {
+  // NEW: Feature-detect availability of additional screen information.
+  if ("getScreens" in window) {
+    try {
+      // NEW: Requests permission-gated access to additional screen information.
+      let screensInterface = await window.getScreens();
+      startMultiScreenSlideshow(screensInterface);
+      return;
+    } catch (err) {
+      // The request was denied or an error occurred.
+      console.error(err.name, err.message);
+    }
+  }
+  // If multi-screen info is not available, start a single-screen slideshow.
+  startSingleScreenSlideshow();
+};
+```
+
+### Add `Screens` and `ScreenAugmented` interfaces for additional screen info
+
+The `getScreens()` method grants access to a `Screens` interface on success.
+That provides multi-screen information and change events, as well as additional
+per-screen information via a `ScreenAugmented` interface, which inherits from
+the existing [`Screen`](https://drafts.csswg.org/cssom-view/#screen) interface.
+The proposed shapes of these interfaces, exposed to secure contexts with
+explicit permission, are outlined below:
+
+```webidl
+// NEW: Interface exposing multiple screens and additional information.
+[SecureContext] interface Screens : EventTarget {
+  // NEW: The set of available screens with additional per-screen info.
+  readonly attribute FrozenArray<ScreenAugmented> screens;
+
+  // NEW: A reference to the current screen with additional info.
+  readonly attribute ScreenAugmented currentScreen;
+
+  // NEW: An event fired when 'screens' or 'currentScreen' changes.
+  // NOTE: Does not fire on changes to attributes of individual Screens.
+  attribute EventHandler onchange;
+};
+
+// NEW: Interface inherits Screen and exposes additional information.
+[SecureContext] interface ScreenAugmented : Screen {
```

Comment:
I think Augmented sounds weird... makes me  think of augmented reality. Maybe we can find a better name

Reply:
ScreenExtendedDetails? Or ScreenAdvancedDetails

Reply:
I changed it to ScreenAdvanced for now, and I'm open to changing the name with additional input here. I'm slightly worried that 'Details' might ostensibly preclude exposing 'control' surfaces on the interface, like the existing [Screen Orientation API](https://www.w3.org/TR/screen-orientation/#extensions-to-the-screen-interface). I'd like this API shape to also be readily extensible if there's eventually interest in offering parallel controls for resolution, mirroring, etc. like existing orientation locks.